src/fsharp/FSharp.Build/Fsc.fs

```diff
@@ -198,17 +199,23 @@ type public Fsc () as this =
         // WarningLevel
         builder.AppendSwitchIfNotNull("--warn:", warningLevel)
 
+        // warnOn
+        let warnOnArray =
+            match warnOn with
+            | null -> [||]
+            | _ -> warnOn.Split([|' '; ';'; ','|], StringSplitOptions.RemoveEmptyEntries)
+
```

Comment:
Results should be trimmed because there could be newlines if a project file does this, which is common:

``` xml
<WarnOn>
$(WarnOn);
123;
456;
</WarnOn>
```